src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java

```diff
@@ -12,30 +13,22 @@
  * @author adrianhopebailie
  *
  */
-public class PreimageSha256Fulfillment implements Fulfillment {
+public class PreimageSha256Fulfillment extends FulfillmentBase {
+
+	public static final ConditionType CONDITION_TYPE = ConditionType.PREIMAGE_SHA256;
+
+	public PreimageSha256Fulfillment(ConditionType type, byte[] payload) {
+		super(type, payload);
+	}
 
-	private static ConditionType CONDITION_TYPE = ConditionType.PREIMAGE_SHA256;
-	
 	private static EnumSet<FeatureSuite> BASE_FEATURES = EnumSet.of(
 			FeatureSuite.SHA_256, 
 			FeatureSuite.PREIMAGE
 		);
 
-	private byte[] preimage;
```

Comment:
I think we have different idea about how to use the fulfillment. I thought the fulfillment should be very flexible so it can even be instantiated with a default constructor and then populated with getters and setters. That allows us to use it in patterns that depend on this like dependency injection although this is perhaps a bad example.

I agree that the condition should be immutable and generated from the fulfillment.